### PR TITLE
feat: integrate storage-initializer rock in track/0.10

### PR DESCRIPTION
**Pull request by NohaIhab**
use rock for storageInitializer built in https://github.com/canonical/kserve-rocks/blob/track/v0.10.0/storage-initializer/rockcraft.yaml
Closes https://github.com/canonical/kserve-operators/issues/198

### CODE REVIEW

**Comment by kimwnasptd** [comment]
LGTM and the image gets pulled. Let's just fix the CI and we can merge

**Comment by misohu** [comment]
Can we update the README

File: README.md [hunk 1]
```diff
@@ -202,29 +202,8 @@ Please see the [official docs site](https://charmed-kubeflow.io/docs) for comple
 
 ## Setting Custom Images for Kserve controller
 
-KServe controller comes with a set of preconfigured images that are used in Kserve workloads. This is the list of default images used in charm.
+KServe controller comes with a set of preconfigured images that are used in Kserve workloads. The list of default images used in the charm can be found in [default-custom-images.json](https://github.com/canonical/kserve-operators/blob/track/0.10/charms/kserve-controller/src/default-custom-images.json).
```
Comment (kimwnasptd) [inline README.md:205]: As @orfeas-k nicely mentioned, we could use relative paths here so that we don't tie ourselves with a branch (or have to update)

Comment (NohaIhab) [reply README.md]: good point, committed

**Review by orfeas-k — APPROVED** [review]
Approved, thank you @NohaIhab.

**Pull request merged**